auth/principal.go

```diff
@@ -145,3 +145,69 @@ type User interface {
 
 	setRolesSince(ch.TimedSet)
 }
+
+// PrincipalConfig represents a user/role as a JSON object.
+// Used to define a user/role within DbConfig, and structures the request/response body in the admin REST API
+// for /db/_user/*
+type PrincipalConfig struct {
+	Name             *string  `json:"name,omitempty"`
+	ExplicitChannels base.Set `json:"admin_channels,omitempty"`
+	// Fields below only apply to Users, not Roles:
+	Email             *string  `json:"email,omitempty"`
+	Disabled          *bool    `json:"disabled,omitempty"`
+	Password          *string  `json:"password,omitempty"`
+	ExplicitRoleNames base.Set `json:"admin_roles,omitempty"`
```

Comment:
I'm not sure, but I think the reason was consistency with `ExplicitChannels`. Can revert it, though if I'm not mistaken there shouldn't be a BWC issue since `base.Set` has a `JSONMarshaler`/`JSONUnmarshaler` that serialises it as an array anyway.

Reply:
That's fine, didn't realise the unmarshaller worked that way. Thx